src/PreserveModuleNamePlugin.ts

```diff
@@ -72,90 +164,305 @@ function getPreservedModules(modules: Webpack.Module[]) {
   return new Set(
     modules.filter(m => {
       // Some modules might have [preserveModuleName] already set, see ConventionDependenciesPlugin.
-      let value = m[preserveModuleName];      
-      for (let r of m.reasons) {
-        if (!r.dependency || !r.dependency[preserveModuleName]) continue;
-        value = true;
-        let req = removeLoaders((r.dependency as ModuleDependency).request);
-        // We try to find an absolute string and set that as the module [preserveModuleName], as it's the best id.
-        if (req && !req.startsWith(".")) {
-          m[preserveModuleName] = req;
-          return true;
-        }
+      if (m[preserveModuleName]) {
+        return true;
```

Comment:
It seems to me that you have significantly changed the behavior of this.
We did not just check if some dependency had `preserveModuleName` set.

If the dependency used an absolute module id, we would use it as a module name, as this is more robust than trying to figure out the normalized id ourselves. See line 82 before PR.

Unless you have a good reason to remove that behavior, I would like to revert this change.

Reply:
I am confused by this. Why would we set a modules id based on one of its dependencies? ModuleA could require ModuleB but we dont want to set ModuleA's id based on ModuleB which might or might not live in the same module directory.

Reply:
Well I said it here:
**If the dependency used an absolute module id**, we would use it as a module name, as **this is more robust than trying to figure out the normalized id ourselves**.

Not sure how to say it better.
Normalizing the module names for aurelia-loader is a can of worms. It works rather well but there are lots of edge cases, this PR is proof of that.
So when we are handed out an absolute module id, that's great because it means we don't have to make guesses at normalization and it's guaranteed to work. 
It's also marginally faster.